integration_testing/connection_controller_test.go

```diff
@@ -382,6 +394,8 @@ func TestConnectionControllerProtocolDifference(t *testing.T) {
 }
 
 func TestConnectionControllerPersistentConnection(t *testing.T) {
+	t.Cleanup(ResetDisableNetworkManagerRoutines)
```

Comment:
good - using t.Cleanup is best practice for cleaning up after your tests. However, I may update `SetDisableNetworkManagerRoutines` to take in `*testing.T` and add the `t.Cleanup` call in there so you don't need to have `t.Cleanup` and then `SetDisableNetworkManagerRoutines` everywhere. You can simply have `SetDisableNetworkManagerRoutines(t)`